utils/efs-mode-parsing.ts

```diff
@@ -0,0 +1,35 @@
+import * as efs from 'aws-cdk-lib/aws-efs';
+
+/**
+ * Parse the performance mode string into an EFS PerformanceMode.
+ * @param performanceMode The performance mode string to parse.
+ * @returns The EFS PerformanceMode.
+ */
+export function parsePerformanceMode(performanceMode: string): efs.PerformanceMode {
+  switch (performanceMode) {
+    case 'GENERAL_PURPOSE':
+      return efs.PerformanceMode.GENERAL_PURPOSE;
+    case 'MAX_IO':
+      return efs.PerformanceMode.MAX_IO;
+    default:
+      throw new Error(`Invalid performance mode: ${performanceMode}`);
+  }
```

Comment:
**Suggestion:** Refactor the `parsePerformanceMode` and `parseThroughputMode` functions to use a mapping object instead of a switch-case structure for better maintainability and scalability. This approach simplifies adding new modes in the future. [maintainability, importance: 8]
```suggestion
const performanceModes = {
  'GENERAL_PURPOSE': efs.PerformanceMode.GENERAL_PURPOSE,
  'MAX_IO': efs.PerformanceMode.MAX_IO
};
if (!performanceModes[performanceMode]) {
  throw new Error(`Invalid performance mode: ${performanceMode}`);
}
return performanceModes[performanceMode];
```